pkg/system/chtimes.go

```diff
@@ -2,14 +2,28 @@ package system
 
 import (
 	"os"
+	"syscall"
 	"time"
+	"unsafe"
 )
 
+// maxTime returns the maximum time that can be set by os.Chtimes on this platform
+// 64 bit Timespec supports up to 2262 and 32 bit Timespec supports up to 2038
+func maxTime() time.Time {
+	if unsafe.Sizeof(syscall.Timespec{}.Nsec) == 8 {
+		// This is a 64 bit timespec
+		// os.Chtimes limits time to the following
+		return time.Unix(0, 1<<63-1)
+	}
+
+	// This is a 32 bit timespec
+	return time.Unix(1<<31-1, 0)
+}
+
 // Chtimes changes the access time and modified time of a file at the given path
 func Chtimes(name string, atime time.Time, mtime time.Time) error {
 	unixMinTime := time.Unix(0, 0)
-	// The max Unix time is 33 bits set
-	unixMaxTime := unixMinTime.Add((1<<33 - 1) * time.Second)
```

Comment:
I was on the edge of the undefined behavior caused by having more than int64 max nanoseconds when using ```.UnixNano()```. 33 bits of seconds resulted in just less than int64 max nanoseconds, and 34 bits of seconds overflowed.

It turns out that Golang's ```os.Chtimes()``` limits it to int64 max (63 bits) nanoseconds (instead of seconds) on 64 bit Timespecs and int32 max seconds on 32 bit Timespecs. This PR fixes my incorrect assumptions about maximum time values.